src/web/web-service/wpan_service.cpp

```diff
@@ -0,0 +1,357 @@
+        break;
+
+    case kWpanStatus_OffLine:
+        networkInfo["WPAN service"] = "Offline";
+        networkInfo["mDNS service"] = "Down";
+        break;
+
+    case kWpanStatus_Down:
+        networkInfo["wpantund"] = "Down";
+        networkInfo["WPAN service"] = "Offline";
+        networkInfo["mDNS service"] = "Down";
+        break;
+
+    default:
+        break;
+    }
+    root["result"] = networkInfo;
+    root["error"] = ret;
+    response = jsonWriter.write(root);
+    return response;
+}
+
+std::string WpanService::HandleAvailableNetworkRequest()
+{
+    Json::Value              root, networks, networkInfo;
+    ot::Dbus::WPANController wpanController;
+    Json::FastWriter         jsonWriter;
+    std::string              response;
+    int                      ret = ot::Dbus::kWpantundStatus_Ok;
+
+    wpanController.SetInterfaceName(mIfName);
+    VerifyOrExit(wpanController.Leave() == ot::Dbus::kWpantundStatus_Ok,
+                 ret = ot::Dbus::kWpantundStatus_LeaveFailed);
+    VerifyOrExit(wpanController.Scan() == ot::Dbus::kWpantundStatus_Ok,
+                 ret = ot::Dbus::kWpantundStatus_ScanFailed);
+    mNetworksCount = wpanController.GetScanNetworksInfoCount();
+    VerifyOrExit(mNetworksCount > 0, ret = ot::Dbus::kWpantundStatus_NetworkNotFound);
+    memcpy(mNetworks, wpanController.GetScanNetworksInfo(),
+           mNetworksCount * sizeof(ot::Dbus::WpanNetworkInfo));
+
+    for (int i = 0; i < mNetworksCount; i++)
+    {
+        char extPanId[OT_EXTENDED_PANID_LENGTH * 2 + 1], panId[OT_PANID_LENGTH + 3],
+             hardwareAddress[OT_HARDWARE_ADDRESS_LENGTH * 2 + 1];
+        ot::Utils::Long2Hex(Thread::Encoding::BigEndian::HostSwap64(mNetworks[i].mExtPanId), extPanId);
+        ot::Utils::Bytes2Hex(mNetworks[i].mHardwareAddress, OT_HARDWARE_ADDRESS_LENGTH, hardwareAddress);
+        sprintf(panId, "0x%X", mNetworks[i].mPanId);
+        networkInfo[i]["nn"] = mNetworks[i].mNetworkName;
+        networkInfo[i]["xp"] = extPanId;
+        networkInfo[i]["pi"] = panId;
+        networkInfo[i]["ch"] = mNetworks[i].mChannel;
+        networkInfo[i]["ha"] = hardwareAddress;
+    }
+    root["result"] = networkInfo;
+exit:
+    if (ret != ot::Dbus::kWpantundStatus_Ok)
+    {
+        root["result"] = mResponseFail;
+        otbrLog(OTBR_LOG_ERR, "Error is %d", ret);
+    }
+    root["error"] = ret;
+    response = jsonWriter.write(root);
+    return response;
+}
+
+int WpanService::GetWpanServiceStatus(std::string &aNetworkName, std::string &aExtPanId) const
+{
+    std::string              wpantundState = "";
+    int                      status = kWpanStatus_OK;
+    ot::Dbus::WPANController wpanController;
+
+    wpanController.SetInterfaceName(mIfName);
+    wpantundState = wpanController.Get(kWPANTUNDProperty_NCPState);
+    if (wpantundState.length() == 0)
+    {
+        status = kWpanStatus_Down;
+        ExitNow();
+    }
+    if (wpantundState == "associated")
```

Comment:
Is it better to list all possible states and define them as cons string or macros?